grails-app/services/org/pih/warehouse/receiving/ReceiptService.groovy

```diff
@@ -287,26 +294,39 @@ class ReceiptService {
         }
     }
 
+    /**
+     We can create RECEIVED or PARTIALLY_RECEIVED event depending on the case:
+        1. When the location supports partial receiving
+            RECEIVED - when the shipment is not already received and is fully received
+            PARTIALLY_RECEIVED - when the shipment wasn't partially received before (it's
+                created only once)
+        2. When the location doesn't support partial receiving
+            RECEIVED - after receiving for the first time, we cannot receive
+                for the second time without partial receiving. The rest of the remaining quantities
+                should be canceled
+            PARTIALLY_RECEIVED - not allowed
+     */
     void savePartialReceiptEvent(PartialReceipt partialReceipt) {
         Shipment shipment = partialReceipt.shipment
         Receipt receipt = partialReceipt.receipt
 
-        if (shipment.isFullyReceived()) {
-            if (!shipment.wasReceived()) {
-                shipmentService.createShipmentEvent(shipment,
-                        receipt?.actualDeliveryDate,
-                        EventCode.RECEIVED,
-                        shipment.destination)
-            }
-        } else {
+        if (!shipment.wasReceived() && (!shipment.destination.supports(ActivityCode.PARTIAL_RECEIVING) || shipment.isFullyReceived())) {
```

Comment:
also, I'm not totally sure about this new flow. The old flow looked correct to me. Mainly:
```
// We're fully receiving
if (fully received now) {
  if (it was not already in fully received status) {
    create fully received event
  }
}
// We're partially receiving
else if (it was not already in partially received status) {
  create partial received event
}
```
so I'm wondering if there's an issue not with this class but with `shipment.isFullyReceived()`. Or maybe I'm just misunderstanding. I haven't had a chance to dig much deeper into the code